system/Database/BaseBuilder.php

```diff
@@ -837,13 +847,21 @@ protected function _whereIn(string $key = null, array $values = null, bool $not
 
 		$not = ($not) ? ' NOT' : '';
 
-		$where_in = array_values($values);
-		$ok       = $this->setBind($ok, $where_in, $escape);
+		if ($values instanceof Closure)
+		{
+			$builder = clone $this;
+			$builder->from([], true)->resetQuery();
```

Comment:
Since we're using this same code in 2 places, we should refactor those out into their own method to make future maintenance simpler

Reply:
made it